### PR TITLE
chore: add tracing for pipeline

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

image

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13428)

### CODE REVIEW

**Comment by BohuTANG** [comment]
Need to check the performance is drop or not when add minitrace to the hot path.

**Comment by andylokandy** [comment]
The ci will run benchmark with tracing is turned off. I suppose this is sufficient.

File: src/query/pipeline/sinks/src/async_mpsc_sink.rs [hunk 1]
```diff
@@ -152,6 +152,7 @@ impl<T: AsyncMpscSink + 'static> Processor for AsyncMpscSinker<T> {
     }
 
     #[async_backtrace::framed]
+    #[minitrace::trace(name = "AsyncMpscSinker::async_process")]
```
Comment (zhang2014) [inline src/query/pipeline/sinks/src/async_mpsc_sink.rs:155]: Maybe move it to https://github.com/datafuselabs/databend/blob/main/src/query/pipeline/core/src/processors/processor.rs#L172 is better?

File: src/query/pipeline/sinks/src/sync_sink.rs [hunk 2]
```diff
@@ -108,6 +108,7 @@ impl<T: Sink + 'static> Processor for Sinker<T> {
         self.inner.interrupt()
     }
 
+    #[minitrace::trace(name = "Sinker::process")]
```
Comment (zhang2014) [inline src/query/pipeline/sinks/src/sync_sink.rs:111]: Maybe move it to https://github.com/datafuselabs/databend/blob/main/src/query/pipeline/core/src/processors/processor.rs#L168 is better?

File: src/query/pipeline/sinks/src/async_mpsc_sink.rs (hunk 1 above)
Comment (andylokandy) [reply src/query/pipeline/sinks/src/async_mpsc_sink.rs:155]: if so, we don't know what exact the processor is running.

**Comment by BohuTANG** [comment]
askbend:summary

**Review by zhang2014 — APPROVED** [review]

**Pull request merged**